### PR TITLE
[C#] Cap incremental memory usage in InputStream

### PR DESCRIPTION
**Pull request by chadwalters**
Prevent InputStream from preallocating large amounts of memory without reading from underlying stream in incremental chunks. Default is set to 128MB to prevent massive preallocation by corrupted data, similar to the DeserializerControls introduced in https://github.com/Microsoft/bond/pull/297

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,9 +16,14 @@ different versioning scheme, following the Haskell community's
...
 * IDL comm version: TBD
 * C++ version: TBD
-* C# NuGet version: TBD
+* C# NuGet version: TBD (minor bump needed)
 * C# Comm NuGet version: TBD (minor bump needed)
 
+## C# ###
+
+* Added controls to cap incremental allocation between reads in
+  IO.UnsafeInputStream.
```
Comment (chwarr) [inline CHANGELOG.md]: The type is `Bond.IO.Unsafe.InputStream`.

File: cs/test/core/StreamTests.cs [hunk 2]
```diff
@@ -88,7 +113,14 @@ public void StreamBufferReuseTest()
         delegate void IntTest<T>(T value);
 
         [Test]
-        public void Varint()
+        public void VarintTest()
```
Comment (chwarr) [inline cs/test/core/StreamTests.cs]: Minor: `VarInt`? It currently looks like a misspelling of variant. :-)

File: cs/src/io/unsafe/InputStream.cs [hunk 3]
```diff
@@ -13,6 +13,31 @@ public class InputStream : InputBuffer, ICloneable<InputStream>
...
+                activeAllocationChunk = value;
+            }
+        }
+
+        static InputStream()
+        {
+            ActiveAllocationChunk = DefaultAllocationChunk;
+        }
+
+        static int activeAllocationChunk;
```
Comment (chwarr) [inline cs/src/io/unsafe/InputStream.cs]: Minor: move static field above member fields.

File: cs/src/io/unsafe/InputStream.cs [hunk 4]
```diff
@@ -13,6 +13,31 @@ public class InputStream : InputBuffer, ICloneable<InputStream>
     {
         readonly Stream stream;
         readonly int bufferLength;
+
+        // Default setting for maximum incremental allocation chunk size before reading from stream
+        public const int DefaultAllocationChunk = 128 * 1024 * 1024;
```
Comment (chwarr) [inline cs/src/io/unsafe/InputStream.cs]: Minor: move const above member fields

**Comment by chadwalters** [comment]
See revision. If it looks good, please squash.

**Comment by chwarr** [comment]
Will do after CI builds finish.

**Comment by chwarr** [comment]
Failures are C++ build timeouts that we're working on addressing. Merging.

**Pull request merged**